### DISCUSSION TITLE
Docker fails after Windows update

### DISCUSSION THREAD

**Discussion by Alvin-Prasla** [microsoft/dotnet-framework-docker#880] [Q&A]
We encountered an issue with Docker when windows updates were recently applied on our Windows Server 2016 box. All builds were working correctly and then below windows updates broke the build. To prove, we uninstalled those windows update and things started to work fine. Because those updates are necessary as per our server admins we installed it back and docker is failing again.

The windows update list which is causing issue:

1. KB4535680
2. KB4589210
3. KB4601392
4. KB5005698
5. KB459824

Dockerfile:
```
# escape=`
FROM microsoft/dotnet-framework:4.7.2-sdk-windowsservercore-ltsc2016
SHELL ["cmd", "/S", "/C"]

RUN Mkdir BuildOctopus
COPY powershell \BuildOctopus
```

After the image is successfully created, I run the image to create the container. Now from inside the container I run (for example) octo.exe to check the version (for simplicity). The command octo.exe --version doesn't return any error, no logs, no output. I tried various other commands also but its same. Then I tried below code to check last error code.

PowerShell:
```
PS C:\BuildOctopus> $LASTEXITCODE = 0
PS C:\BuildOctopus> $LASTEXITCODE
0
PS C:\BuildOctopus> .\octo.exe --version
PS C:\BuildOctopus> $LASTEXITCODE
-1073741502
```
Basically octo.exe doesn’t respond from within the docker container (Yes the octo.exe path is correct and I verified the file exists by visually seeing using dir command and also wrote a small PS script to check if the file exists). This worked fine before the updates were installed. The octo.exe also works fine outside the container.

- Host Edition: Windows server 2016 Standard
- Version: 1607
- Octopus.exe: 3.3.18 (I know its old, the concern is why it worked before windows update and failed after updates were applied.)

Stack overflow: [https://stackoverflow.com/questions/70760419/docker-fails-after-windows-update](https://stackoverflow.com/questions/70760419/docker-fails-after-windows-update)

**Comment by mthalman** [comment]
Are you able to narrow down which specific KB update is causing this behavior?

What is Octopus and where does it come from? Is it this: https://octopus.com/?

I'm assuming that installation of the Build Tools is not needed in order to reproduce the failure. Can you confirm that?

@weijuans-msft - Something for you to take a look at.

**Reply by Alvin-Prasla** [reply]
The list is already narrow down. At some point our server admins were unable to uninstall certain updates.

Yes, Its the Octopus deploy command line exe and is currently inside BuildOctopus directory. The link to exact version [Octo_3.3.18](https://octopus.com/downloads/octopuscli/3.3.18)

I updated the dockerfile to the correct version which is failing and narrowed it further.

**Reply by Alvin-Prasla** [reply]
Any update please?
